src/components/Home.jsx

```diff
@@ -1,19 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
 import getToken from '../lib/tokens';
 import { useHistory } from 'react-router-dom';
+import db from '../lib/firebase';
+import firebase from 'firebase/app';
+
+import './Home.css';
 
 const Home = () => {
   let history = useHistory();
+  const [tokenName, setTokenName] = useState('');
+  const [notification, setNotification] = useState(null);
 
   const generateToken = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+
+    db.collection('tokens')
+      .doc('tokenDocument')
+      .update({
+        tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
+      })
+      .catch((error) => {
+        db.collection('tokens')
+          .doc('tokenDocument')
+          .set({
+            tokenArray: [token],
+          });
+      });
+
     history.push('/list-view');
   };
 
+  const resetError = () => {
+    setTimeout(() => {
+      setNotification(null);
+    }, 10000);
+  };
```

Comment:
The error Zachee pointed out is showing because of this line. It means that if I get that error and decide to go elsewhere, like if we had other pages and I decide to click elsewhere upon getting that error, the timer will keep reading in the background even though I had left the component. Now React is recommending you put the asynchronous task (settimeout) inside of useEffect, so that you can cancel the timer when a user decides to leave the component.

This is how I implemented it locally and it clears the error:

Mere introducing the useEffect hook clears the error:

`  const [error, setError] = useState(false);`

`  useEffect(() => {
    if (error) {
      setTimeout(() => {
        setError(false);
      }, 1000);
    }
  }, [error])`

How I used it:

        `} else {
          setError(true)
          setNotification(
            'Token not found - please check spelling and try again or create new list',
          );
        }
      })
      .catch((error) => {
        setError(true)
        setNotification(
          'An unexpected error occurred, please refresh the page and try again',
        );
      });`

But if you want to implement the React best practices to fullest then we introduce the return statement of useEffect for clearing sideEffects:

``` let timerId;

  const resetError = () => {
    timerId = setTimeout(() => setError(false), 1000);
  };


  useEffect(() => {
    if (error) resetError()
    return () => {
      clearTimeout(timerId);
    };
  }, [error])```


//This clears the error, if you decide to implement this let me know if you have more questions🙂

Reply:
Wow this is really helpful, thank you so much! 🙏🏻